src/server/master_server/master_metadata_service_impl.cc

```diff
@@ -19,13 +26,90 @@ inline server::MetadataManager* MasterMetadataServiceImpl::metadata_manager() {
   return server::MetadataManager::GetInstance();
 }
 
+inline server::ChunkServerManager& MasterMetadataServiceImpl
+    ::chunk_server_manager() {
+  return server::ChunkServerManager::GetInstance();
+}
+
 grpc::Status MasterMetadataServiceImpl::HandleFileCreation(
     const protos::grpc::OpenFileRequest* request,
     protos::grpc::OpenFileReply* reply) {
+  // Step 1. Create file metadata
   const std::string& filename(request->filename());
   google::protobuf::util::Status status(
       metadata_manager()->CreateFileMetadata(filename));
-  return common::utils::ConvertProtobufStatusToGrpcStatus(status);
+  if (!status.ok()) {
+    return common::utils::ConvertProtobufStatusToGrpcStatus(status);
+  }
+
+  // Step 2. Create the first file chunk for this file
+  google::protobuf::util::StatusOr<std::string> chunk_or(
+      metadata_manager()->CreateChunkHandle(filename,0));
+  if (!chunk_or.ok()) {
+    return common::utils::ConvertProtobufStatusToGrpcStatus(status);
+  }
+  const std::string& chunk(chunk_or.ValueOrDie());
```

Comment:
nit. it'd be really readable if you call this `chunk_handle`; i kept thinking this is file chunk proto or something when reading code

Reply:
Thanks for pointing this out, I revised it.